nixos/modules/virtualisation/nixos-containers.nix

```diff
@@ -713,15 +717,21 @@ in
 
   config = mkIf (config.boot.enableContainers) (let
 
+    warnings = flatten [
+      (optional (config.virtualisation.containers.enable && versionOlder config.system.stateVersion "20.09") ''
+        Enabling both boot.enableContainers & virtualisation.containers on system.stateVersion < 20.09 is unsupported.
```

Comment:
I'm very sorry if I forgot about something we discussed in the past, but I'm quite unhappy with this since it's generally a bad idea to bump your system's `stateVersion`. This basically means that each system which was provisioned before 20.09 cannot use `virtualisation.containers`.

Wouldn't it be possible to add e.g. an option `stateDir` for `nixos-containers` which is `mkDefault "/var/lib/containers"` on `stateVersion<20.09` and `mkDefault "/var/lib/nixos-containers"` on `stateVersion<=20.09`?

Then you could warn if `/var/lib/containers` is used and an admin could alter `statePath` as soon as all containers are migrated. If that (optional) migration is done, both tools would be usable on one system.

Reply:
> I'm very sorry if I forgot about something we discussed in the past, but I'm quite unhappy with this since it's generally a bad idea to bump your system's

This PR doesn't introduce this incompatibility, the incompatibility _already_ exists. What I'm doing right there is making users aware of a dangerous configuration that is already unsupported.

I'm reluctant to add more options at the top-level of this module, especially as it doesn't have any namespacing, but instead introduces new arguments in the `boot` attribute.